specification/deviceupdate/resource-manager/Microsoft.DeviceUpdate/preview/2021-11-02-preview/deviceupdate.json

```diff
@@ -0,0 +1,1845 @@
+            "$ref": "./examples/PrivateEndpointConnectionProxies/PrivateEndpointConnectionProxy_Delete.json"
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location"
+        }
+      }
+    },
+    "/providers/Microsoft.DeviceUpdate/operations": {
+      "get": {
+        "description": "Returns list of operations for Microsoft.DeviceUpdate resource provider.",
+        "operationId": "Operations_List",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Operation details.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/OperationListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing the reason for operation failure.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "deprecated": false,
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    }
+  },
+  "definitions": {
+    "Account": {
+      "description": "Device Update account details.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/TrackedResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "description": "Device Update account properties.",
+          "x-ms-client-flatten": true,
+          "type": "object",
+          "properties": {
+            "provisioningState": {
+              "description": "Provisioning state.",
+              "type": "string",
+              "enum": [
+                "Succeeded",
+                "Deleted",
+                "Failed",
+                "Canceled",
+                "Accepted",
+                "Creating"
+              ],
+              "readOnly": true,
+              "x-ms-enum": {
+                "name": "ProvisioningState",
+                "modelAsString": true
+              }
+            },
+            "hostName": {
+              "description": "API host name.",
+              "type": "string",
+              "readOnly": true
+            },
+            "publicNetworkAccess": {
+              "description": "Whether or not public network access is allowed for the container registry.",
+              "default": "Enabled",
+              "enum": [
+                "Enabled",
+                "Disabled"
+              ],
+              "type": "string",
+              "x-ms-enum": {
+                "name": "PublicNetworkAccess",
+                "modelAsString": true
+              }
+            },
+            "sku": {
```

Comment:
>sku [](http://example.com/codeflow?start=13&length=3)

Shouldn't sku be at at the top level? Instead of under properties?

[https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md)

Reply:
This is copied over from the existing API version, I had followed the instructions of copying over the existing API version and then adding in the new changes. So as this is existing, i really dont want to have to change it.

Reply:
Which existing api version is that?

I don't see SKU property anywhere in

[https://github.com/Azure/azure-rest-api-specs/blob/main/specification/deviceupdate/resource-manager/Microsoft.DeviceUpdate/preview/2020-03-01-preview/deviceupdate.json](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/deviceupdate/resource-manager/Microsoft.DeviceUpdate/preview/2020-03-01-preview/deviceupdate.json)

Reply:
I am sorry, I commented on the wrong place. You are correct this did not come from an older version. This is intentional for now, we would like the sku in the properties, and if we need it on the top level, we will do it later before we go GA.

Reply:
@StanleyOverton Why not just fix it now? It only gets harder to make this kind of change the longer you leave it, in my experience. And a new api-version while your service is in preview should be the best time to make such a change.

Reply:
Because we dont need it to be in the top level at least as far as we can see. What is the issue that you see if we leave it here? Are we loosing something if we leave it here?

Reply:
@TimLovellSmith Can you please tell me what the gain if from moving this to the top level? We have this working the way we want and dont foresee any issues with the way it is. We would like to know why this would be an issue?

Reply:
Fine question. The biggest thing I remember you _gaining_ is that Azure Resource Graph supports caching sku only if it is at the top level. (That might be old information, sorry if it is.) You are also gaining compliance with Azure Rest API spec and ARM guidelines which are supposed to be in the readiness criteria for public preview i.e. its not supposed to be optional whether you follow the spec or not. (And what are you really gaining by postponing the fix?) This might not be a complete list. I'll update it if I think of more.

I can imagine you might also be potentially losing out on ability to do integrations like autoscale and quota management if you literally never fix this, since they might assume you have a top level 'sku' property, following the conventions like properties called 'capacity', but I haven't really done the research on that yet.

Reply:
@TimLovellSmith @StanleyOverton 
I found this in documentation - SKU is a top-level property:
https://armwiki.azurewebsites.net/api_contracts/RPCstrongvalidation.html
Stan, please follow up with RPaaS because they may have a different validation. If they OK it, we should add this to our Resource as an optional top level property.